apps/src/lib/node/ledger/shell/prepare_proposal/block_space_alloc/states/protocol_txs.rs

```diff
@@ -0,0 +1,87 @@
+use std::marker::PhantomData;
+
+use super::super::{AllocStatus, BlockSpaceAllocator, TxBin};
+use super::{
+    BuildingEncryptedTxBatch, BuildingProtocolTxBatch, NextStateImpl, State,
+    WithEncryptedTxs, WithoutEncryptedTxs,
+};
+
+impl State for BlockSpaceAllocator<BuildingProtocolTxBatch> {
+    #[inline]
+    fn try_alloc(&mut self, tx: &[u8]) -> AllocStatus {
+        // TODO: prioritize certain kinds of protocol txs;
+        // this can be done at the `CheckTx` level,
+        // we don't need the `TxBin`s to be aware
+        // of different prioriy hints for protocol txs
```

Comment:
Yeah this will be it's own issue. I don't think we can prioritize at the `CheckTx` level in Tendermint v0.37, so we can only prioritize the protocol txs given to use by Tendermint within the context of an individual `RequestPrepareProposal`.